### PR TITLE
Trend scanning labels

### PR DESCRIPTION
**Pull request by proskurin**
# Description

Add Trend-Scanning labels implementation

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] test_labelling.py

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: mlfinlab/tests/test_labels.py [hunk 1]
```diff
@@ -292,3 +298,33 @@ def test_drop_labels(self):
         # Assert that threshold works
         new_labels = drop_labels(events=triple_labels, min_pct=0.20)
         self.assertTrue(0 in set(new_labels['bin']))
+
+    def test_trend_scanning_labels(self):
```
Comment (Jackal08) [inline mlfinlab/tests/test_labels.py]: About how long does this take to run? 

There are actually a couple of unit tests in here. Should we not rather have them in their own test method with a docstring explaining what they test for?

Comment (Jackal08) [reply mlfinlab/tests/test_labels.py]: I am nervous that it adds another few minutes to our already long build times.

File: mlfinlab/tests/test_labels.py [hunk 2]
```diff
@@ -27,6 +28,11 @@ def setUp(self):
         self.data = pd.read_csv(self.path, index_col='date_time')
         self.data.index = pd.to_datetime(self.data.index)
 
+        # data set used for trend scanning labels
+        self.stock_prices = pd.read_csv('test_data/stock_prices.csv', index_col=0, parse_dates=[0])
```
[the remark this answers is not in the document]
Comment (proskurin) [reply mlfinlab/tests/test_labels.py]: It is already used by HRP tests, so the file is already there

File: mlfinlab/tests/test_labels.py (hunk 1 above)
Comment (proskurin) [reply mlfinlab/tests/test_labels.py]: Let me check how long the test takes

Comment (proskurin) [reply mlfinlab/tests/test_labels.py]: The test takes 1s ;)

File: docs/source/implementations/labeling_trend_scanning.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
+.. _implementations-labeling_trend_scanning:
+
+==========================================
+Labeling: Trend Scanning
+==========================================
+
+Trend scanning is a classification labeling technique introduced by Marcos Lopez de Prado in the following lecture slides: "Advances in Financial Machine Learning: Lecture 3/10"
+'https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3257419'
```
Comment (Jackal08) [inline docs/source/implementations/labeling_trend_scanning.rst:8]: This url can be added as a link. Please see the PortOpt modules for a good example. Its better to have a link of a word than a long url.

File: mlfinlab/labeling/trend_scanning.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+"""
+
+import pandas as pd
+import numpy as np
+from mlfinlab.structural_breaks.sadf import _get_betas
+
+
+def get_trend_scanning_labels(price_series: pd.Series, t_events: list, look_forward_window: int) -> pd.DataFrame:
+    """
+    Get Trend-Scanning labels (https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3257419)
```
[the remark this answers is not in the document]
Comment (Jackal08) [reply mlfinlab/labeling/trend_scanning.py:14]: Sorry I know I said I would do this but I see that your documentation already answers it. Would you mind to please add it?

File: mlfinlab/tests/test_labels.py [hunk 5]
```diff
@@ -11,11 +11,12 @@
 from mlfinlab.filters.filters import cusum_filter
 from mlfinlab.labeling.labeling import add_vertical_barrier, get_events, get_bins, drop_labels
 from mlfinlab.util.volatility import get_daily_vol
+from mlfinlab.labeling import trend_scanning_labels
 
 
 class TestChapter3(unittest.TestCase):
```
Comment (Jackal08) [inline mlfinlab/tests/test_labels.py:17]: The trend scanning labels should have their own class. They are not part of Chapter 3.

File: mlfinlab/tests/test_labels.py [hunk 6]
```diff
@@ -11,11 +11,12 @@
...
     """
-    Test Triple barrier, meta-labeling, dropping rare labels, and daily volatility.
+    Test Triple barrier, meta-labeling, trend-scanning, dropping rare labels, and daily volatility.
```
Comment (Jackal08) [inline mlfinlab/tests/test_labels.py:19]: not part of ch3

**Pull request merged**